include/crocoddyl/multibody/contacts/contact-1d.hxx

```diff
@@ -90,9 +100,23 @@ void ContactModel1DTpl<Scalar>::updateForce(const boost::shared_ptr<ContactDataA
                  << "lambda has wrong dimension (it should be 1)");
   }
   Data* d = static_cast<Data*>(data.get());
-  const Eigen::Ref<const Matrix3s> R = d->jMf.rotation();
-  data->f.linear() = R.col(2) * force[0];
-  data->f.angular() = d->jMf.translation().cross(data->f.linear());
+  pinocchio::SE3 jMc;
+  switch (type_) {
+    case pinocchio::LOCAL: {
+      jMc = d->jMf;
+      break;
+    }
+    case pinocchio::WORLD: {
+      jMc = d->jMf.act(d->pinocchio->oMf[id_].inverse());
+      break;
+    }
+    case pinocchio::LOCAL_WORLD_ALIGNED: {
+      jMc = pinocchio::SE3(Matrix3s::Identity(), d->jMf.translation());
```

Comment:
This is allocating data dynamically.
Instead, you should modify the translation only, i.e.
```suggestion
 jMc_.translation = d->jMf.translation();
```

Then, you have to set the identity matrix only when you have define `LOCAL_WORLD_ALIGNED`